### PR TITLE
Move inherents to primitives

### PR DESCRIPTION
**Pull request by gnunicorn**
Next step towards #4099, this PR removes any dependency from `client` to `paint`. This is being achieved by moving all inherents into primitives, creating three more super thin crates, that all others then reference. The newly added files, therefore, are just code moves with minor modifications regarding import names.

The new crates are prefixed already with `sp-` as per the convention that will be following.

### CODE REVIEW

File: client/consensus/aura/src/lib.rs [hunk 1]
```diff
@@ -654,9 +660,9 @@ fn register_aura_inherent_data_provider(
 	inherent_data_providers: &InherentDataProviders,
 	slot_duration: u64,
 ) -> Result<(), consensus_common::Error> {
-	if !inherent_data_providers.has_provider(&paint_aura::INHERENT_IDENTIFIER) {
+	if !inherent_data_providers.has_provider(&aura_primitives::inherents::INHERENT_IDENTIFIER) {
```
Comment (tomusdrw) [inline client/consensus/aura/src/lib.rs]: Can't we just import it directly? I find it a bit confusing to have both `use aura_primitives::...` at the top and then use full path to import some other stuff (I know the other one is `pub use`, but still the full path here feels a bit too long).

File: primitives/timestamp/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+[package]
+name = "sp-timestamp"
+version = "2.0.0"
+authors = ["Parity Technologies <admin@parity.io>"]
+edition = "2018"
+
+[dependencies]
+sr-api = { path = "../sr-api", default-features = false }
+rstd = { package = "sr-std", path = "../sr-std", default-features = false }
```
Comment (tomusdrw) [inline primitives/timestamp/Cargo.toml]: no renames?

Comment (gnunicorn) [reply primitives/timestamp/Cargo.toml]: I mostly kept the existing names and copied `Cargo.toml` files to keep the changes low. I would fix this in the upcoming PR if you don't mind.

Comment (tomusdrw) [reply primitives/timestamp/Cargo.toml]: Sure!

**Comment by gavofyork** [comment]
looks good - merge when ready @gnunicorn

**Pull request merged**